src/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -9,18 +9,34 @@ namespace System.Diagnostics {
   public partial class Activity {
     public Activity(string operationName) {}      
     public string OperationName { get { throw null; } }
+
+    [System.ObsoleteAttribute]
     public string Id {get { throw null; } private set {} }
+
     public DateTime StartTimeUtc {get { throw null; } private set {} }
     public Activity Parent {get { throw null; } private set {} }
-    public string ParentId {get { throw null; } private set {} }
-    public string RootId {get { throw null; } private set {} }    
+  
+    public string ParentId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string RootId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string SpanId { get { throw null; } private set { } }
+
+    public byte TraceFlags {get { throw null; } set {} }
```

Comment:
SpanID and TraceFlags are basically 'converters' for W3C format.    In theory they are not essential (you can parse the ID yourself), but having them can help making things efficient.     

I have no problem with having a Recorded Boolean property on Activity (get and SetRecorded method), as that is a useful function of activities (to remember if you are an 'Important' activity).  

I am suspicious that having TraceFlags is not really needed, as long as the rules for dealing with IDs do the right thing (propagate the flags).    The more the system deals only with IDs and not the parts the better.